HubSpot.NET/Api/Engagement/HubSpotEngagementApi.cs

```diff
@@ -34,96 +33,68 @@ public EngagementHubSpotModel Create(EngagementHubSpotModel entity)
         /// <returns>The engagement</returns>
-        public EngagementHubSpotModel GetById(long engagementId)
-        {
-            var path = $"{new EngagementHubSpotModel().RouteBasePath}/engagements/{engagementId}";
-
-            var data = _client.Execute<EngagementHubSpotModel>(path, Method.GET, false);
-            return data;
-        }
+        public EngagementHubSpotModel GetById(long engagementId) 
+            => _client.Execute<EngagementHubSpotModel>(GetRoute<EngagementHubSpotModel>(engagementId.ToString()));
 
         /// <summary>
         /// Retrieves a paginated list of engagements
         /// </summary>
         /// <param name="opts">Options for querying</param>
         /// <returns>List of engagements, with additional metadata, e.g. total</returns>
-        public EngagementListHubSpotModel<T> List<T>(EngagementListRequestOptions opts = null) where T: EngagementHubSpotModel, new()
+        public EngagementListHubSpotModel<T> List<T>(EngagementListRequestOptions opts = null) where T: EngagementHubSpotModel
         {
-            if (opts == null)
-            {
-                opts = new EngagementListRequestOptions();
-            }
+            opts =  opts ?? new EngagementListRequestOptions();
 
-            var path = $"{new EngagementHubSpotModel().RouteBasePath}/engagements/paged".SetQueryParam("limit", opts.Limit);
+            var path = $"{GetRoute<T>("paged")}".SetQueryParam("limit", opts.Limit);
 
-            if (opts.Offset.HasValue)
-            {
-                path = path.SetQueryParam("offset", opts.Offset);
-            }
+            if (opts.Offset.HasValue)            
+                path = path.SetQueryParam("offset", opts.Offset);            
 
-            var data = _client.ExecuteList<EngagementListHubSpotModel<T>>(path, opts, convertToPropertiesSchema: false);
-            return data;
+            return _client.Execute<EngagementListHubSpotModel<T>, EngagementListRequestOptions>(path, opts);            
         }
 
         /// <summary>
         /// Lists recent engagements (i.e. in date order)
         /// </summary>
         /// <param name="opts">Options for querying</param>
         /// <returns>List of engagements, with additional metadata, e.g. total</returns>
-        public EngagementListHubSpotModel<T> ListRecent<T>(EngagementListRequestOptions opts = null) where T : EngagementHubSpotModel, new()
+        public EngagementListHubSpotModel<T> ListRecent<T>(EngagementListRequestOptions opts = null) where T : EngagementHubSpotModel
         {
-            if (opts == null)
-            {
-                opts = new EngagementListRequestOptions();
-            }
-            var path = $"{new EngagementHubSpotModel().RouteBasePath}/engagements/recent/modified".SetQueryParam("count", opts.Limit);
-
-            if (opts.Offset.HasValue)
-            {
+            opts = opts ??new EngagementListRequestOptions();
```

Comment:
Not fussed right now, but I wonder if it's worth getting some automatic linting / style guide in?